include/dlaf/factorization/cholesky/impl.h

```diff
@@ -36,72 +36,81 @@ namespace factorization {
 namespace internal {
 
 namespace cholesky_l {
-template <class Executor, Device device, class T>
-void potrfDiagTile(Executor&& exec, hpx::future<matrix::Tile<T, device>> matrix_tile) {
-  hpx::dataflow(exec, matrix::unwrapExtendTiles(tile::potrf_o), blas::Uplo::Lower,
-                std::move(matrix_tile));
+template <Backend backend, class MatrixTileSender>
+void potrfDiagTile(MatrixTileSender&& matrix_tile) {
+  // TODO:
+  // - when_all(...) | algo(policy) | detach()?
+  // - algo(policy, when_all(...)) | detach?
+  // - detach(algo(policy, when_all(...)))?
```

Comment:
Up for discussion: which style do you prefer?

Reply:
The second style is more similar to what we are used to, I think. 
But the first one looks easier to understand, in my opinion!

Reply:
I think the 1st one is the more "linear" one, at least in the way I use to think the DAG.

Reply:
I agree with both of you, but I just want to recommend that if you haven't already done so you should also look at the real examples below for comparison. `potrfDiagTile` now uses the second option, and the ones below the first option. The three examples above are a bit deceivingly simple when the actual arguments are missing and your opinion *might* change from those.

I personally like the second one. It's a bit closer to `dataflow(exec/policy, algo, ...)` with the change that `...` is wrapped in `whenAllLift`. The `detach` I consider a completely separate operation so I like having it behind the `|`. In the end this is "just" syntax and not the end of the world, but readability matters as well. On the other hand, this is one of the easier things to change afterwards.

Reply:
Yes, we get used to dataflow, but it doesn't mean that they looked like the most readable things 😛


The 1st one, as already said, is more linear; it's a real pipeline, given inputs, you perform an action. If I try to find a possible downside w.r.t. 2nd option, in the latter one the function name is more exposed, being the first thing you read in the line. However, I also think that it is not so hidden in the 1st one (but it may require some manual formatting in some cases).

I'm ok with both, as you said, it's not a big deal (I wouldn't consider the 3rd one, it's too nested for me 😄).

Reply:
I agree with @albestro. It is true that the 2nd is closer to `dataflow`, but I've always found dataflow a bit hard to read. Even if the function name is after |  in the 1st option, this still looks easier to me! 
By the way, both 1st and 2nd options to me are perfectly fine! :)

Reply:
I'm a bit more in favor of option 2... but option 1 is not bad either.

Reply:
Looks like 1.5 votes for 1. and 2. I personally also prefer 2. because the algorithm name is more obvious.

Reply:
Option 1 it is.